### PR TITLE
PXP-572: Handle blue/green deployment failure on deployment

### PR DESCRIPTION
**Pull request by mfauzaan**
## Summary

Currently with an application that is having Blue-Green deployment topology, we don’t have an effective method to guarantee that the Blue deployment is running BEFORE sending the Green deployment however Mindvalley does not follow this practice yet so need additional handling to check for the deployment status, and ask for confirmation if there is any failure.

Ticket: [https://mindvalley.atlassian.net/browse/PXP-572](https://mindvalley.atlassian.net/browse/PXP-572)

## What's Changed

- [x]  Added additional confirmation if there is failure on last deployment

#### CASE 1: BLUE deployment is succeeded
Screenshot 2023-08-01 at 4 03 03 PM

#### CASE 2: BLUE deployment is not  succeeded
Screenshot 2023-08-01 at 4 01 43 PM

### CODE REVIEW

**Review by onimsha — CHANGES_REQUESTED** [review]
There are two things from this PR:

* It doesn't cover the case of an application that doesn't have the `blue` version. If it's the case, the CLI simply throwing an error from the API like this
```
❯ ./target/debug/wukong deployment execute
Not detecting any flags, entering deployment terminal......
Current application: mv-wukong-api-proxy
✔ Step 1: Please choose the namespace you want to deploy · Prod
You've selected `Prod` as the deployment namespace.

✔ Step 2: Please choose the version you want to deploy · Green
You've selected `Green` as the deployment version.

✔ Step 3: Checking the status of the latest Blue deployment...
Error - API Response Error: cdPipeline:3:3: cd_pipelines_version_config_not_defined
```
We should handle this, and allow the deployment to proceed if there's no blue deployment

* Also, we should have test case to cover this logic. If there's test case we will be able to detect the flaw in the logic.

**Comment by mfauzaan** [comment]
> There are two things from this PR:
> 
> * It doesn't cover the case of an application that doesn't have the `blue` version. If it's the case, the CLI simply throwing an error from the API like this
> 
> ```
> ❯ ./target/debug/wukong deployment execute
> Not detecting any flags, entering deployment terminal......
> Current application: mv-wukong-api-proxy
> ✔ Step 1: Please choose the namespace you want to deploy · Prod
> You've selected `Prod` as the deployment namespace.
> 
> ✔ Step 2: Please choose the version you want to deploy · Green
> You've selected `Green` as the deployment version.
> 
> ✔ Step 3: Checking the status of the latest Blue deployment...
> Error - API Response Error: cdPipeline:3:3: cd_pipelines_version_config_not_defined
> ```
> 
> We should handle this, and allow the deployment to proceed if there's no blue deployment
> 
> * Also, we should have test case to cover this logic. If there's test case we will be able to detect the flaw in the logic.

Handle the case where by there is only one version of the deployment which ignores the check
For second point we still have the limitation of doing the test cases for interactive sessions

**Comment by jk-gan** [comment]
Screenshot 2023-08-02 at 3 22 44 PM

Step 2 suddenly jump to Step 4 in this case

**Review by jk-gan — APPROVED** [review]
LGTM

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/127#pullrequestreview-1558442634" title="Approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/127#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by onimsha — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/127#pullrequestreview-1558442634" title="Approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/127#" title="Author self-approved">mfauzaan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/127#pullrequestreview-1558584650" title="Approved">onimsha</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
